internal/logger/io_sink.go

```diff
@@ -7,91 +7,43 @@
 package logger
 
 import (
+	"encoding/json"
 	"io"
-	"log"
+	"time"
 )
 
-// IOSink writes to an io.Writer using the standard library logging solution and
-// is the default sink for the logger, with the default IO being os.Stderr.
+// IOSink writes a JSON-encoded message to the io.Writer.
 type IOSink struct {
-	log *log.Logger
+	enc *json.Encoder
 }
 
-// Compile-time check to ensure osSink implements the LogSink interface.
+// Compile-time check to ensure IOSink implements the LogSink interface.
 var _ LogSink = &IOSink{}
 
-// NewIOSink will create a new IOSink that writes to the provided io.Writer.
+// NewIOSink will create an IOSink object that writes JSON messages to the
+// provided io.Writer.
 func NewIOSink(out io.Writer) *IOSink {
 	return &IOSink{
-		log: log.New(out, "", log.LstdFlags),
+		enc: json.NewEncoder(out),
```

Comment:
Something we lose when we don't use the `log` package is buffered writing. Should we wrap `out` in a `bufio.Writer` and flush the buffer at the end of calls to `Info`/`Error`?

Reply:
What benefit would this add? I don't think we would actually use the buffer since we would flush every time Info/Error is called.

Reply:
As discussed separately, looks like a `json.Encoder` maintains its own internal buffer and writes the entire message at once, so we don't need to add any write buffer.